pkg/services/service_example_configured_test.go

```diff
@@ -0,0 +1,102 @@
+package services_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+
+	. "github.com/smartcontractkit/chainlink-common/pkg/internal/example"
+)
+
+type configured struct {
+	services.Service
+	internals *services.Internals
+
+	subA services.Service
+	subB services.Service
+
+	workCh chan func() (name string, err error)
+}
+
+func (e *configured) start(context.Context) error {
+	e.internals.GoRepeat(time.Minute, e.do)
+	return nil
+}
+
+// do processes all outstanding work
+func (e *configured) do() {
+	for {
+		select {
+		case <-e.internals.StopChan:
+			return
+		case work, ok := <-e.workCh:
+			if !ok {
+				return
+			}
+			name, err := work()
+			if err != nil {
+				e.internals.SetUnhealthy(name, err)
+			} else {
+				e.internals.SetHealthy(name)
+			}
+		default:
+			return
+		}
+	}
+}
+
+func newFakeService(lggr logger.Logger, name string) services.Service {
+	return services.NewService(lggr, name)
+}
+
+func NewConfigured(lggr logger.Logger) services.Service {
+	e := &configured{
+		workCh: make(chan func() (string, error)),
+	}
+	e.Service, e.internals = services.Config{
+		Name:  "Configured",
+		Start: e.start,
+		NewSubServices: func(lggr logger.Logger) []services.Service {
+			e.subA = newFakeService(lggr, "Sub-service-A")
+			e.subB = newFakeService(lggr, "Sub-service-B")
+			return []services.Service{e.subA, e.subB}
+		},
+	}.NewService(lggr)
+	return e
+}
+
+func ExampleConfig_NewService() {
+	lggr, err := Logger()
+	if err != nil {
+		fmt.Println("Failed to create logger:", err)
+		return
+	}
+	s := NewConfigured(lggr)
+	if err = s.Start(context.Background()); err != nil {
+		fmt.Println("Failed to start service:", err)
+		return
+	}
+	if err = s.Close(); err != nil {
+		fmt.Println("Failed to close service:", err)
+		return
+	}
+
+	// Output:
```

Comment:
Is there a txtar like way to test against this output? That might be too brittle since the sub services are closed concurrently.

Reply:
> Is there a txtar like way to test against this output?

It is already built-in. These execute like regular tests.
https://go.dev/blog/examples

> That might be too brittle since the sub services are closed concurrently.

We can probably add a test-only hack to make the close deterministic :thinking:

Reply:
I think there is still some non-determinism in this example

Reply:
"Fixed" by commenting out (the output) with a note so it does not run in the test suite. It can still be invoked from the docs site.